client/app/pages/queries/QuerySource.jsx

```diff
@@ -1,19 +1,128 @@
-import React from 'react';
+import { find, map, clone } from 'lodash';
+import React, { useState, useRef, useEffect, useMemo, useCallback } from 'react';
 import PropTypes from 'prop-types';
 import { react2angular } from 'react2angular';
+import Select from 'antd/lib/select';
 import { routesToAngularRoutes } from '@/lib/utils';
 import { Query } from '@/services/query';
+import { DataSource, SCHEMA_NOT_SUPPORTED } from '@/services/data-source';
+import notification from '@/services/notification';
+import recordEvent from '@/services/recordEvent';
 
 import QueryPageHeader from './components/QueryPageHeader';
+import SchemaBrowser from './components/SchemaBrowser';
+
+import './query-source.less';
+
+function getSchema(dataSource, refresh = undefined) {
+  if (!dataSource) {
+    return Promise.resolve([]);
+  }
+
+  return dataSource.getSchema(refresh)
+    .then((data) => {
+      if (data.schema) {
+        return data.schema;
+      } else if (data.error.code === SCHEMA_NOT_SUPPORTED) {
+        return [];
+      }
+      return Promise.reject(new Error('Schema refresh failed.'));
+    })
+    .catch(() => {
+      notification.error('Schema refresh failed.', 'Please try again later.');
+      return Promise.resolve([]);
+    });
+}
+
+function QuerySource(props) {
+  const [query, setQuery] = useState(props.query);
+  const [dataSources, setDataSources] = useState([]);
+  const [dataSourcesLoaded, setDataSourcesLoaded] = useState(false);
+  const dataSource = useMemo(() => (find(dataSources, { id: query.data_source_id }) || null), [query, dataSources]);
```

Comment:
Need to find a way to handle this for the `new` page :slightly_smiling_face: 

Otherwise...
![image](https://user-images.githubusercontent.com/3356951/70623767-8e855500-1bfd-11ea-93bc-d5d19354af1e.png)

The behavior on the Angular page is to use `localStorage.lastSelectedDataSourceId || first(dataSources)`

Reply:
Thanks for pointing. ATM I'm testing both pages, so probably will find even more issues.

Reply:
Fixed :+1: Thank you once more!